extends CharacterBody2D

@export var velocidad: float = 500
var direccion: Vector2 = Vector2.ZERO

func _ready():
	$AnimatedSprite2D.play("default")

	# Destruir el proyectil despu칠s de 5 segundos
	await get_tree().create_timer(5.0).timeout
	if is_inside_tree():
		queue_free()

func _physics_process(delta):
	if direccion != Vector2.ZERO:
		# Normaliza la direcci칩n y aplica la velocidad
		velocity = direccion.normalized() * velocidad
		move_and_slide()

		# 游댃 Rotar el sprite hacia la direcci칩n del disparo
		# Si tu sprite est치 dibujado mirando hacia la derecha, deja esta l칤nea as칤
		rotation = direccion.angle()
		
		# Si est치 dibujado mirando hacia arriba, usa esto en su lugar:
		# rotation = direccion.angle() + deg_to_rad(90)

		# Animaci칩n de vuelo
		$AnimatedSprite2D.play("default")

		# Colisiones
		for i in range(get_slide_collision_count()):
			var collider = get_slide_collision(i).get_collider()

			if collider.is_in_group("jugador"):
				queue_free()
			elif collider.is_in_group("mapa"):
				queue_free()
			elif collider.is_in_group("enemigo1") or collider.is_in_group("enemigo2"):
				continue
